src/common/plugin/PluginState.ts

```diff
@@ -248,6 +248,20 @@ export function makePluginState(settings: PluginStateSettings): PluginState {
         return
       }
 
+      // Force enableCustomServers to false server list is empty because
+      // an empty server list would cause the wallets to never sync, and
+      // this is unlikely the user's intention.
+      // This means policy change was decided later, which makes
+      // `enableCustomServers` functionally only useful for disabling custom
+      // servers entirely even when passing a list of servers (which could have
+      // been done by passing an empty list to begin with).
+      // In other words, this field is more of an internal field to track whether
+      // to treat the list a static or not (whether to fetch serve lists outside
+      // of the list provided by the user or not).
+      const enableCustomServers =
+        newSettings.enableCustomServers &&
+        newSettings.blockbookServers.length === 0
```

Comment:
I can't comprehend this logic. The `&&` is really tripping me up. I fee like It would make sense if the second term was `!== 0`, but this way seems backwards. Has this been tested? What am I missing here?

Reply:
I believe this is a valid catch. I had originally had it as a tertiary operator and simplified to a conjunction and missed that I needed to invert the condition from `===` to `!==`.